File: runtime/dart/ui/textDecoration.ts
import { IList } from "../core/list";
declare const dart: {
    ui: {
        textDecorationCombine: (
            decorations: IList<ITextDecoration>
        ) => ITextDecoration;
        textDecorationNone: (this: void) => ITextDecoration;
        textDecorationUnderline: (this: void) => ITextDecoration;
        textDecorationOverline: (this: void) => ITextDecoration;
        textDecorationLineThrough: (this: void) => ITextDecoration;
    };
};
export interface ITextDecoration {
    contains: (other: ITextDecoration) => boolean;
    getHashCode: () => number;
    toString: () => string;
}
export class TextDecoration {
    public static none = dart.ui.textDecorationNone();
    public static underline = dart.ui.textDecorationUnderline();
    public static overline = dart.ui.textDecorationOverline();
    public static lineThrough = dart.ui.textDecorationLineThrough();
    public static combine(
        decorations: IList<ITextDecoration>
    ): ITextDecoration {
        return dart.ui.textDecorationCombine(decorations);
    }
    private readonly _dart_contains: (other: ITextDecoration) => boolean =
        undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    public contains(other: ITextDecoration): boolean {
        return this._dart_contains(other);
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
    public toString(): string {
        return this._dart_toString();
    }
}

File: lib/cfr/builtins/libs/dart/ui/textDecoration.dart
import 'dart:core';
import 'dart:ui';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedTextDecoration extends VMManagedBox<TextDecoration> {
  VMManagedTextDecoration(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['contains'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.contains(maybeUnBoxAndBuildArgument<TextDecoration, dynamic>(
            luaCallerArguments[1],
            parentState: hydroState)),
      ];
    });
    table['getHashCode'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.hashCode,
      ];
    });
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final TextDecoration vmObject;
}

void loadTextDecoration(
    {required HydroState hydroState, required HydroTable table}) {
  table['textDecorationNone'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<TextDecoration>(
          object: TextDecoration.none,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['textDecorationUnderline'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<TextDecoration>(
          object: TextDecoration.underline,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['textDecorationOverline'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<TextDecoration>(
          object: TextDecoration.overline,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['textDecorationLineThrough'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<TextDecoration>(
          object: TextDecoration.lineThrough,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['textDecorationCombine'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<TextDecoration>(
          object: TextDecoration.combine(
              maybeUnBoxAndBuildArgument<List<TextDecoration>, TextDecoration>(
                  luaCallerArguments[1],
                  parentState: hydroState)),
          hydroState: hydroState,
          table: HydroTable()),
    ];
  });
  registerBoxer<TextDecoration>(boxer: (
      {required TextDecoration vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedTextDecoration(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}
